aioftp/client.py

```diff
@@ -603,9 +796,81 @@ class AsyncClientLister(AsyncListerMixin):
         :return: path info
         :rtype: :py:class:`dict`
         """
-        code, info = await self.command("MLST " + str(path), "2xx")
-        name, info = self.parse_mlsx_line(str.lstrip(info[1]))
-        return info
+        await self.ensure_cwd()
+
+        if type(path) is str:
+
+            path = pathlib.PurePosixPath(path)
+
+        if self.list_compatibility_level == 2:
+
+            try:
+
+                code, info = await self.command("MLST " + str(path), "2xx")
+
+            except errors.StatusCodeError as e:
+
+                if e.received_codes[-1] == "500":
+
+                    self.list_compatibility_level = 1
+
+                else:
+
+                    return None
+
+            else:
+
+                name, info = self.parse_mlsx_line(str.lstrip(info[1]))
+
+                return info
```

Comment:
You don't need `else` branch here, you can paste `else`-branch code right after `MLST`, since if we got `5xx`, then code below `command` will not be executed.